src/main/java/org/spongepowered/common/registry/util/RegistryModuleLoader.java

```diff
@@ -97,6 +100,21 @@ private static boolean hasCatalogRegistration(RegistryModule module) {
                 return true;
             }
         }
+        if (module instanceof AlternateCatalogRegistryModule) {
+            try {
+                final Type type = module.getClass().getGenericSuperclass();
+                Class<?> catalogClass = (Class<?>) ((ParameterizedType) type).getActualTypeArguments()[0];
+                CatalogedBy catalogedBy = catalogClass.getAnnotation(CatalogedBy.class);
```

Comment:
What is this actually doing? Why are you checking for the `CatalogedBy` annotation?